src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt

```diff
@@ -144,10 +146,16 @@ abstract class FixupSession(
   private fun ensureSelectionRange(agent: CodyAgent, textFile: ProtocolTextDocument) {
     val selection = textFile.selection ?: return
     selectionRange = selection.toRangeMarker(document, true)
-    agent.server
-        .getFoldingRanges(GetFoldingRangeParams(uri = textFile.uri, range = selection))
-        .thenApply { result -> selectionRange = result.range.toRangeMarker(document, true) }
-        .get()
+
+    try {
+      val result =
+          agent.server
+              .getFoldingRanges(GetFoldingRangeParams(uri = textFile.uri, range = selection))
+              .get()
+      selectionRange = result.range.toRangeMarker(document, true)
+    } catch (e: Exception) {
+      logger.warn("Error getting folding range", e)
```

Comment:
Does that mean that instead of failing we will be now showing selection edit lenses in possibly wrong place? (I'm not saying it's wrong decision, just asking about the end result ;))

Reply:
This is really funny!  Ah, me. I hope we can get this fixed with tree-sitter very soon, so that it just never happens.